lib/rules/semi-spacing.js

```diff
@@ -117,33 +117,51 @@ module.exports = {
         }
 
         /**
+         * Report location example :
+         *
+         * for unexpected space `before`
+         *
+         * var a = 'b'   ;
+         *            ^^^
+         *
+         * for unexpected space `after`
+         *
+         * var a = 'b';  c = 10;
+         *             ^^
+         *
          * Reports if the given token has invalid spacing.
          * @param {Token} token The semicolon token to check.
          * @param {ASTNode} node The corresponding node of the token.
          * @returns {void}
          */
         function checkSemicolonSpacing(token, node) {
             if (astUtils.isSemicolonToken(token)) {
-                const location = token.loc.start;
 
                 if (hasLeadingSpace(token)) {
+                    const tokenBefore = sourceCode.getTokenBefore(token);
+                    const loc = {
+                        start: tokenBefore.loc.end,
+                        end: token.loc.start
+                    };
+
                     if (!requireSpaceBefore) {
                         context.report({
                             node,
-                            loc: location,
+                            loc,
                             messageId: "unexpectedWhitespaceBefore",
                             fix(fixer) {
-                                const tokenBefore = sourceCode.getTokenBefore(token);
 
                                 return fixer.removeRange([tokenBefore.range[1], token.range[0]]);
                             }
                         });
                     }
                 } else {
                     if (requireSpaceBefore) {
+                        const loc = token.loc;
```

Comment:
It would be nice to add `endLine` and `endColumn` to one test to confirm this change as well

(the same applies to the other `const loc = token.loc;` line).